File: desktop/syntax_stories_desktop/src/lib/tokenStorage.ts
const K_ACCESS = 'syntax_stories_desktop_access';
const K_REFRESH = 'syntax_stories_desktop_refresh';

export const tokenStorage = {
  getAccess(): string | null {
    return localStorage.getItem(K_ACCESS);
  },
  getRefresh(): string | null {
    return localStorage.getItem(K_REFRESH);
  },
  setTokens(access: string, refresh?: string | null): void {
    localStorage.setItem(K_ACCESS, access);
    if (refresh) localStorage.setItem(K_REFRESH, refresh);
    else localStorage.removeItem(K_REFRESH);
  },
  clear(): void {
    localStorage.removeItem(K_ACCESS);
    localStorage.removeItem(K_REFRESH);
  },
};

File: mobile/lib/services/token_storage.dart
import 'package:shared_preferences/shared_preferences.dart';

class TokenStorage {
  static const _kAccess = 'auth_access_token';
  static const _kRefresh = 'auth_refresh_token';

  Future<void> saveTokens({required String access, String? refresh}) async {
    final p = await SharedPreferences.getInstance();
    await p.setString(_kAccess, access);
    if (refresh != null) {
      await p.setString(_kRefresh, refresh);
    }
  }

  Future<String?> readAccess() async {
    final p = await SharedPreferences.getInstance();
    return p.getString(_kAccess);
  }

  Future<String?> readRefresh() async {
    final p = await SharedPreferences.getInstance();
    return p.getString(_kRefresh);
  }

  Future<void> clear() async {
    final p = await SharedPreferences.getInstance();
    await p.remove(_kAccess);
    await p.remove(_kRefresh);
  }
}
